test/src/test/java/org/corfudb/runtime/view/StreamViewTest.java

```diff
@@ -464,4 +466,69 @@ public void testPreviousWithStreamCheckpoint() throws Exception {
     }
+
+    /**
+     * Ensure that transaction stream semantics are correct with respect to different
+     * combinations of trim points and seek positions.
+     *
+     * @throws InterruptedException
+     */
+    @Test
+    public void txLogTrim() throws InterruptedException {
+        final CorfuRuntime.CorfuRuntimeParameters params = CorfuRuntime.CorfuRuntimeParameters
+                .builder()
+                .backpointersDisabled(true)
+                .build();
+        final CorfuRuntime localRuntime = CorfuRuntime.fromParameters(params)
+                .setTransactionLogging(true)
+                .parseConfigurationString(getDefaultConfigurationString())
+                .connect();
+        final CorfuTable<String, String>
+                instance1 = localRuntime.getObjectsView().build()
+                .setTypeToken(new TypeToken<CorfuTable<String, String>>() {})
+                .setStreamName("txTestMap")
+                .open();
+
+        // Populate the Transaction Stream up to NUM_ITERATIONS_LOW entries.
+        IntStream.range(0, PARAMETERS.NUM_ITERATIONS_LOW).forEach(idx -> {
+            localRuntime.getObjectsView().TXBegin();
+            instance1.put(String.valueOf(idx), String.valueOf(idx));
+            localRuntime.getObjectsView().TXEnd();
+        });
+
+        // Force a prefix trim.
+        final long epoch = 0L;
+        localRuntime.getAddressSpaceView().prefixTrim(
+                new Token(epoch, PARAMETERS.NUM_ITERATIONS_LOW/3));
+
+        // Wait until log is trimmed.
+        TimeUnit.SECONDS.sleep(5);
+        localRuntime.getAddressSpaceView().invalidateServerCaches();
+        localRuntime.getAddressSpaceView().invalidateClientCache();
+
+        IStreamView txStream = localRuntime.getStreamsView()
+                .get(ObjectsView.TRANSACTION_STREAM_ID);
+
+        // If the current pointer is below the trim point, we should see the TrimmedException.
+        assertThatThrownBy(() -> txStream.remaining()).isInstanceOf(TrimmedException.class);
+
+        // Seek the transaction stream beyond the trim point. We should not see any issues.
+        txStream.seek(PARAMETERS.NUM_ITERATIONS_LOW/2);
+        txStream.remaining();
+
+        // Populate the transaction stream with additional NUM_ITERATIONS_LOW entries.
+        IntStream.range(0, PARAMETERS.NUM_ITERATIONS_LOW).forEach(idx -> {
+            localRuntime.getObjectsView().TXBegin();
+            instance1.put(String.valueOf(idx), String.valueOf(idx));
+            localRuntime.getObjectsView().TXEnd();
+        });
+
+        // Force a prefix trim beyond our current pointer.
+        localRuntime.getAddressSpaceView().prefixTrim(new Token(epoch,
+                PARAMETERS.NUM_ITERATIONS_LOW + PARAMETERS.NUM_ITERATIONS_LOW/2));
+
+        // Wait until log is trimmed.
+        TimeUnit.SECONDS.sleep(5);
+        assertThatThrownBy(() -> txStream.remaining()).isInstanceOf(TrimmedException.class);
```

Comment:
Can you complete this test with the case I described earlier, seeing that retrying will fail again.. (cause the pointer does not move) and seek again to the end to see we get out of this failing loop.

Reply:
Sounds good.

Reply:
Done.